homeassistant/components/sensor/foobot.py

```diff
@@ -0,0 +1,147 @@
+                          timeout=TIMEOUT)
+    data = {}
+    dev = []
+    try:
+        devices = yield from client.get_devices()
+        _LOGGER.debug("The following devices were found: %s", devices)
+        for device in devices:
+            for sensor_type in SENSOR_TYPES.keys():
+                if sensor_type != 'time':
+                    foobot_sensor = FoobotSensor(client, data, device,
+                                                 sensor_type)
+                    dev.append(foobot_sensor)
+    except (aiohttp.client_exceptions.ClientConnectorError,
+            asyncio.TimeoutError, client.ClientError):
+        _LOGGER.exception('Failed to connect to foobot servers.')
+        raise PlatformNotReady
+    async_add_devices(dev, True)
+
+
+class FoobotSensor(Entity):
+    """Implementation of a Foobot sensor."""
+
+    def __init__(self, client, data, device, sensor_type):
+        """Initialize the sensor."""
+        self._client = client
+        self._uuid = device['uuid']
+        self._name = 'Foobot {} {}'.format(device['name'],
+                                           SENSOR_TYPES[sensor_type][0])
+        self.type = sensor_type
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._last_request = datetime.utcfromtimestamp(0)
+        self._data = data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return SENSOR_TYPES[self.type][2]
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        try:
+            data = self._data[self._uuid][self.type]
+        except(KeyError, TypeError):
+            data = None
+        return data
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the latest data."""
+        if self._uuid in self._data:
+            _LOGGER.debug('last_update: %s', self._last_request)
+            _LOGGER.debug('now: %s', datetime.utcnow())
+            if self._last_request + SCAN_INTERVAL > datetime.utcnow():
+                # If the last request made is recent enough,
+                # just skip the update
+                return
+
+        interval = SCAN_INTERVAL.total_seconds()
+        try:
+            response = yield from self._client.get_last_data(self._uuid,
+                                                             interval,
+                                                             interval + 1)
+        except (aiohttp.client_exceptions.ClientConnectorError,
+                asyncio.TimeoutError, self._client.ClientError):
+            _LOGGER.exception('Failed to fetch data from foobot servers.')
+            raise PlatformNotReady
```

Comment:
*Suggestion:* I know raising `PlatformNotReady` is useful for `async_setup_platform`, but the error is not very descriptive and a `HomeAssistantError` might make more sense here.

Reply:
Good point, I kept using `PlatformNotReady` for temporary errors during setup and changed it for `HomeAssistantError` for everything else.